EventFiltering/PWGEM/EMPhotonFilter.cxx

```diff
@@ -0,0 +1,281 @@
+// Copyright 2019-2020 CERN and copyright holders of ALICE O2.
+// See https://alice-o2.web.cern.ch/copyright for details of the copyright holders.
+// All rights not expressly granted are reserved.
+//
+// This software is distributed under the terms of the GNU General Public
+// License v3 (GPL Version 3), copied verbatim in the file "COPYING".
+//
+// In applying this license CERN does not waive the privileges and immunities
+// granted to it by virtue of its status as an Intergovernmental Organization
+// or submit itself to any jurisdiction.
+
+// \brief software trigger for EM photon
+// \author daiki.sekihata@cern.ch
+
+#include "Math/Vector4D.h"
+#include "Framework/runDataProcessing.h"
+#include "Framework/AnalysisTask.h"
+#include "Framework/AnalysisDataModel.h"
+#include "Framework/ASoAHelpers.h"
+#include "Common/DataModel/CaloClusters.h"
+#include "DataFormatsPHOS/TriggerRecord.h"
+#include "PWGEM/PhotonMeson/DataModel/gammaTables.h"
+#include "PWGEM/PhotonMeson/Core/V0PhotonCut.h"
+#include "PWGEM/PhotonMeson/Core/CutsLibrary.h"
+#include "EventFiltering/filterTables.h"
+#include "Framework/HistogramRegistry.h"
+
+using namespace o2;
+using namespace o2::soa;
+using namespace o2::aod;
+using namespace o2::framework;
+using namespace o2::framework::expressions;
+
+using MyCollisions = soa::Join<aod::Collisions, aod::EvSels>;
+using MyCollision = MyCollisions::iterator;
+
+using MyPrimaryElectrons = soa::Join<aod::EMPrimaryElectrons, aod::EMPrimaryElectronsPrefilterBit>;
+using MyPrimaryElectron = MyPrimaryElectrons::iterator;
+
+struct EMPhotonFilter {
+
+  enum EM_Filter_PhotonType {
+    kPCM = 0x1,
+    kPHOS = 0x2,
+    kEMC = 0x4,
+  };
+
+  enum trigs {
+    kPHOS_Photon = 0,
+    kPHOS_El = 1,
+    kPHOS_Pair = 2,
+    kPHOS_Nbar = 3,
+    kPCM_MatCalib = 4,
+    kPCM_EtaDalitz = 5,
+    kNtrg
+  };
+
+  Produces<aod::PhotonFilters> tags;
+
+  Configurable<float> ePhot{"ePhot", 2.2, "Minimal photon energy (GeV)"};
+  Configurable<float> eEl{"eEl", 1., "Minimal electron energy (GeV)"};
+  Configurable<float> ePair{"ePair", 0.35, "Minimal photon pair mass (GeV)"};
+  Configurable<int> nNbar{"nNbar", 2, "Minimal number of nbar clusters"};
+
+  // for PCM
+  Configurable<float> min_pt_tagging{"min_pt_tagging", 0.f, "max. pT for tagging"};
```

Comment:
should it be min instead of max in the comment?

Reply:
thank you. I will fix it in the next iteration.